### ISSUE TITLE
Text renders outside box

### ISSUE THREAD

**Issue by tibetiroka** [endless-sky/endless-sky#5833]
**Describe the bug**
In some cases text input fields render their contents outside of their visual borders.

**To Reproduce**
Just spam random characters, and eventually you might see the characters appear outside of the box. Usually when I add or remov any character the rendering switches back to normal, but when I re-enter the deleted character it breaks again.

Example input (the exact character codes might differ, I don't know what charset is used in the game):
With special characters:
!!@@#%#$^$%%^^%&(^)O(())(((&(
With a-z characters:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaiiiiiiiiiiiiiiiiiiiiiiii
I just found out that I can enter up to 4 of any character after this input and it will still be rendered outside of the box, or I can enter a dozen i's until it starts working properly. Interestingly, I could not reproduce the issue with the use of 'i' only.

Since the size of the text boxes might depend on the screen size (not sure) it is possible that you may not be able to reproduce the issue with these exact input sequences.

**Expected behavior**
Text should not be rendered outside of the input field

**Screenshots**
![Image showing the issue (no plugins installed)](https://user-images.githubusercontent.com/68112292/114179104-dd044a00-993e-11eb-91b8-264657517b13.png)

**System (please complete the following information):**
 - OS: Windows 10 pro version 20H2 build 19042.867
 - Game Source: Steam
 - Version: 0.9.12 (latest Steam version)
 - Keyboard: Hungarian (QWERTZ, see [this link](http://kbdlayout.info/KBDHU/) for exact layout)

**Comment by tibetiroka** [comment]
I can reproduce it with other text input fields as well:
![image](https://user-images.githubusercontent.com/68112292/114183581-1ee3bf00-9944-11eb-8593-1322965e6dad.png)
Input: 
aaaaaaaaaaaaaaaaaaaaaaaaaIIIIIIIIIIIIIIIiiiiiiiiiiiiiiiiiiiiiii

**Comment by MCOfficer** [comment]
Dupes #3917
TL;DR: The rendering system will be redone anyways for unicode support, so there's no point in putting effort into a very niche visual bug.

**Comment by tehhowch** [comment]
All content creators are responsible for proofing their content, including dialogs, plugin "about" text, etc.

As MCOfficer mentioned, this is a known issue and will be addressed in the future by a change to font rendering implementation and eventual improvements to the ui, e.g. fixed-height modals with scroll bars for vertical overflow. If we ever choose to handle horizontal overflow, it will likely only be to clip the render as-is.

In general, if someone wants to type a garbage word to try to break something, congrats, they will get what they expect.

**Comment by tibetiroka** [comment]
See #8466.

**Comment by bene-dictator** [comment]
Is this closed with #8466?

**Comment by tibetiroka** [comment]
No, the last comment in the pr said it still failed with one of my edge cases

**Comment by Amazinite** [comment]
This is still possible.
![image](https://github.com/endless-sky/endless-sky/assets/17688683/9ca1a259-4e27-45a5-95e3-99d45ee449a7)

**Comment by tibetiroka** [comment]
I can get the text to render outside of the box, but can't the game to crash. Can you reliably reproduce it?

**Comment by Amazinite** [comment]
Re: crashing the game. #11286
